src/System.Buffers/tests/ArrayPool/UnitTests.cs

```diff
@@ -211,5 +215,118 @@ public static void RentingAllBuffersAndCallingRentAgainWillAllocateBufferAndRetu
             Assert.NotNull(rented1);
             Assert.NotNull(rented2);
         }
+
+        /*[Fact]
+        public static void ReturningARentedBufferTwiceThrowsAnException()
+        {
+            using (TestEventListener listener = new TestEventListener("20b30044-b729-457e-8dda-8b41b5dd02e6", EventLevel.Verbose))
+            {
+                ArrayPool<byte> pool = ArrayPool<byte>.Create(maxArrayLength: 16, numberOfArrays: 2);
+                bool eventFired = false;
+
+                listener.RunWithCallback(null, () =>
+                {
+                    // We need to rent two buffers because double returning with only 1 buffer rented means
+                    // we will just drop the second return since we wouldn't have space to pool it anyway
+                    byte[] rented1 = pool.Rent(16);
+                    byte[] rented2 = pool.Rent(16);
+                    pool.Return(rented1);
+                    Assert.Throws<InvalidOperationException>(() => pool.Return(rented1));
```

Comment:
@stephentoub, because I need a debugger attached for this test (or a way to mock the IsAttached property), should I look into using some reflection-foo to do this or open an issue to track how to do things like the framework-wide?